kf_release_maker/release_maker.py

```diff
@@ -0,0 +1,276 @@
+
+        print(f'Remaining PRs: {pr_df.shape[0]}')
+
+        # Extract PR emojis, add them into own column
+        def get_emoji(title):
+            parts = title.split(' ', 2)
+            if len(parts) < 2:
+                return config.EMOJI_NOT_FOUND
+            else:
+                return parts[0]
+        pr_df['emoji'] = pr_df['title'].apply(
+            lambda title: get_emoji(title))
+
+        # Categorize PR emojis, add categories into own column
+        pr_df['category'] = pr_df['emoji'].apply(
+            lambda emoji: config.EMOJI_CATEGORIES.get(
+                emoji,
+                config.EMOJI_CATEGORIES['default'])
+        )
+
+        return pr_df
+
+    def next_release_version(self, prev_version, version_type):
+        """
+        Get next release version based on prev version. Version # uses
+        semantic versioning
+
+        version_type must be one of {'major', 'minor', 'patch'}
+        """
+        if version_type not in config.VALID_VERSION_TYPES:
+            raise ValueError(
+                f'Invalid version type: {version_type}! Version type '
+                f'must be one of {pformat(config.VALID_VERSION_TYPES)}!'
+            )
+
+        parts = [int(p) for p in prev_version.split('.')]
+        if version_type == 'major':
+            new_version = f'{parts[0]+1}.0.0'
+        elif version_type == 'minor':
+            new_version = f'{parts[0]}.{parts[1]+1}.0'
+        else:
+            new_version = f'{parts[0]}.{parts[1]}.{parts[2]+1}'
+
+        print(f'Next release version is {new_version}')
+
+        return new_version
+
+    def create_release_markdown(self, pr_df):
+        """
+        Create release note markdown from PR DataFrame and
+        Write to a markdown file named: `<repo name>-<release version>.md`
+        """
+        print('Creating release note markdown ...\n')
+
+        # Emoji count summary markdown string
+        emoji_counts = ' '.join([
+            f'{idx} x{row["number"]}'
+            for idx, row in pr_df.groupby('emoji').count().iterrows()
+        ])
+        # Category count summary markdown string
+        endpoint = f'{self.base_url}/labels'
+        category_counts = ' '.join([
+            f'[{cat}]({endpoint}/{cat}) x{row["number"]}'
+            for cat, row in pr_df.groupby('category').count().iterrows()
+        ])
+        # PR summary markdown string
+        prs = '\n'.join(
+            [f'- (#{row["number"]}) {row["title"]} - @{row["user"]}'
+             for _, row in pr_df.iterrows()]
+        )
+
+        # Construct release notes markdown
+        title = (
+            'Kids First' + ' '.join(self.repo.split('-')).title().lstrip('Kf')
+        )
+        notes = f"# {title} Release {self.release_version}"
+        notes += '\n\n## Features\n\n'
+        notes += '### Summary\n\n'
+        notes += f'Feature Emojis: {emoji_counts} \n'
+        notes += f'Feature Categories: {category_counts} \n\n'
```

Comment:
What was wrong with using the labels here?

Reply:
Not all of our PRs have labels and it seems like we already use emojis to do some categorization of changes. I decided to bucket the emojis into categories and use those instead.  Also, I think we should talk about how we use emojis and labels and how to avoid redundancy

Reply:
Which prs do not have labels? All PRs that I'm familiar with do and it's also mentioned in the [handbook](https://kids-first.github.io/kf-developer-handbook/guides/pull_requests.html).
Emojis do often correlate with the label, but not all the time.